### PR TITLE
Remove ppx runtime linkage

### PR DESCRIPTION
**Pull request by avsm**
- Remove PPX as a runtime link dependency, to reduce
  binary size.

Reported on http://lists.xenproject.org/archives/html/mirageos-devel/2016-05/msg00011.html

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,3 +7,4 @@ env:
   - secure: "BJobYIRLnVumkSlSwhTLLCFl0Li3IdpIoiuTkC39JJQzk6fyz/g86A56GXsENqwOCmWIARuZbc7F2UEMeV5Cd01P99DZdv9OC3fQUzgFEd7s+9AjYtFh91jKGEtsgC67gCxFayB9WkLiME9tkgDRNKvLRk1/9eXo776js8XG4FA="
   matrix:
   - OCAML_VERSION=latest PACKAGE=vchan DEPOPTS="mirage-xen xen-gnt xen-evtchn"
+  - OCAML_VERSION=4.03.0 PACKAGE=vchan DEPOPTS="mirage-xen xen-gnt xen-evtchn"
```
Comment (hannesm) [inline .travis.yml]: I'd make this `latest` here explicit... to be 4.02 (which is actually not latest anymore d'oh) ;)

File: _oasis [hunk 2]
```diff
@@ -40,6 +42,7 @@ Library vchan_lwt_unix
                       xen-gnt.unix, xenstore, xenstore.client,
                       xenstore_transport, xenstore_transport.lwt,
                       mirage-types.lwt, vchan
+  CSources:           vchan_stubs.c
```
Comment (hannesm) [inline _oasis]: how comes? they're already in library vchan above!?!?

**Comment by talex5** [comment]
This fixed it for me - thanks!

**Comment by hannesm** [comment]
worth checking other packages recently converted to ppx /cc @djs55 (do we have a list of these?)

**Comment by talex5** [comment]
The new Docker tests seem to be broken.

**Comment by djs55** [comment]
Fixed as part of #105

**Pull request closed without merging**